Source/Classes/VSTeamProcess.ps1

```diff
@@ -4,24 +4,46 @@ class VSTeamProcess {
    [string]$URL = $null
    [string]$Description = $null
    [string]$Name = $null
+   [string]$ReferenceName = $null
+   [bool]$IsEnabled = $true
    [bool]$IsDefault = $false
    [string]$Type = $null
-
+   [string]$ProcessTemplate = $null
+   [string]$ParentProcessTypeId = $null
    VSTeamProcess (
       [object]$obj
    )  {
-      $this.ID = $obj.id
-      $this.URL = $obj.url
-      $this.IsDefault = $obj.isDefault
-      $this.Name = $obj.name
-      $this.Type = $obj.type
-
+      $this.AddTypeName('Team.Process')
+      $this.Name                    = $obj.name
+      $this.ProcessTemplate         = $obj.name
+      #Allow for processes from .../process/processes with ID property, or ...work/processes with TypeID
+      if ($obj.PSObject.Properties['typeId']) {
+         $this.ID                      = $obj.typeId
+         $this.URL                     = (_getInstance) + "/_apis/work/processes/" + $obj.typeId
```

Comment:
Why are these URLs being hand built and not using the url property of the object returned?

Reply:
There are two branches to the URI tree where you can get processes either under work/processes or Process/processes. 
process/processes is read only, uses "ID" for the ID,  includes the URL but doesn't give information about which processes are disabled and which one is the default. 
So I switched it to work/processes which calls the ID "typeID", _doesn't include the URL_ (which I need later when setting thinks like work item types in a process, or process behaviours , or process properties), but has the advantage of returning the same, richer, object which Set- and Add- return.      
If only I knew, say, a PM on that team who could have a word with people and tell them to "stick to one name for IDs and include the xxx URL!" 
For now the only way to put the URL back is ugly.

Reply:
hmm. I might ping one of the PMs and see what is going one.